### DISCUSSION TITLE
ECC Curve25519 key can't be uploaded on a keyserver (PGP global directory/Symantec Encryption Verified Directory)

### DISCUSSION THREAD

**Discussion by martgil** [openpgpjs/openpgpjs#1552] [Q&A] [answered]
Hello OpenPGP.js team,

Greetings!

I'd like to learn about ECC keys generated by OpenPGP.js and is it true that ECC keys are in V3 format?

I'm trying to upload an ECC curve25519 key to the following keyserver but it forbids the key to be uploaded.
<img width="907" alt="Screen Shot 2022-08-10 at 1 59 56 PM" src="https://user-images.githubusercontent.com/46025304/183828538-3898047e-a3e4-4166-9491-775902907426.png">

I also tried uploading an RSA key (4096 bits) and the keyserver accepts it without any problem.

Is it a known limitation of an ECC key or will there be an option in the key generation function to use V4 format? I'm completely new to PGP/OpenPGP scene but would like to learn about it as much as possible.

Keyservers that (probably) don't support ECC keys:
http://keys.dstl.gov.uk/vkd/GetWelcomeScreen.event
https://keyserver.pgp.com/vkd/GetWelcomeScreen.event

Thank you!
Mart

**Comment by larabr** [comment, marked as answer]
Hi @martgil , we do not generate v3 keys. The problem is that PGP global directory probably doesn't support ECC keys, since it's a very old keyserver. I'd recommend looking into [keys.openpgp.org](http://keys.openpgp.org/) instead.

**Reply by martgil** [reply]
Oh, okay, sir! Thanks for the help @larabr and have a nice day!
